pkg/storage/iterators.go

```diff
@@ -191,12 +254,35 @@ func (f *filteredTupleKeyIterator) Stop() {
 	f.iter.Stop()
 }
 
+// Head returns the next most tuple in the underlying iterator that meets
+// the filter function this iterator was constructed with.
+// Note: the underlying iterator for unmatched filter may advance until head is satisfied.
```

Comment:
```suggestion
// Note: the underlying iterator for unmatched filter may advance until filter is satisfied.
```

also... oof. This kind of breaks my mental model that Head was supposed to not move the iterator 😓